src/db/car/many.rs

```diff
@@ -64,14 +65,29 @@ impl<WriterT> ManyCar<WriterT> {
         );
     }
 
-    pub fn with_read_only_files(self, files: impl Iterator<Item = PathBuf>) -> io::Result<Self> {
+    pub fn with_read_only_files(
+        self,
+        files: impl Iterator<Item = PathBuf>,
+    ) -> anyhow::Result<Self> {
         self.read_only_files(files)?;
         Ok(self)
     }
 
-    pub fn read_only_files(&self, files: impl Iterator<Item = PathBuf>) -> io::Result<()> {
+    pub fn read_only_files(&self, files: impl Iterator<Item = PathBuf>) -> anyhow::Result<()> {
+        let mut cars = vec![];
         for file in files {
```

Comment:
This does not sort the existing CARs though. How about refactoring `read_only` field using [`BinaryHeap`](https://doc.rust-lang.org/stable/std/collections/struct.BinaryHeap.html)

Reply:
Good idea.